src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -7,10 +8,48 @@ package dsls.regex
  * to *remove* anything that currently appears in the file.
  */
 
+ object regexImplicits {
+ 	implicit def charToLiteral(c : Char): RegularExpression =
+ 		Literal(c)
+ 	
+ 	implicit def stringToRegularExpression(s: String): RegularExpression = {
+ 		var firstConcat = Concat(Literal(s.charAt(0)),Literal(s.charAt(1)))
+
+ 		s.substring(2).toList.foldLeft(firstConcat) { (result, element) =>
+ 							   		Concat(result, Literal(element))
+ 							   	}
+ 	}
+ }
+
+
 /** The top of a class hierarchy that encodes regular expressions. */
 abstract class RegularExpression {
-  /** returns true if the given string matches this regular expression */
-  def matches(string: String) = RegexMatcher.matches(string, this)
+  	/** returns true if the given string matches this regular expression */
+  	def matches(string: String) = RegexMatcher.matches(string, this)
+
+	def ||(other: RegularExpression) = {
+		Union(this, other)
+	}
+
+	def ~(other: RegularExpression) = {
+		Concat(this, other)
+	}
+
+	def <*> = {
+		Star(this)
+	}
+
+	def <+> = {
+		Concat(this, Star(this))
+	}
+
+	def apply(n: Integer) = {
+		var regularExpr = this
```

Comment:
This can also be expressed as `this ~ this{n-1}` (with a base case of `EPSILON`.